mathics/core/symbols.py

```diff
@@ -170,8 +170,7 @@ def is_machine_precision(self) -> bool:
 
     def get_lookup_name(self):
         "Returns symbol name of leftmost head"
-
-        return self.get_name()
+        return ""
```

Comment:
In master, BaseExpression returns `self.get_name()`. This requires at least one function call that at the end is useless: you cannot associate definitions to `Atom`s  that are not `Symbol`s